java/sage/media/sub/SubtitleHandler.java

```diff
@@ -341,15 +372,22 @@ else if (sage.media.format.MediaFormat.VOBSUB.equals(targetFormat))
   // Returns true if the timing delay for the next subtitle update has changed (i.e. VF needs a kick)
   public boolean postSubtitleInfo(long time, long dur, byte[] rawText, int flags)
   {
-    synchronized (subEntries)
+    if ((flags & FLUSH_SUBTITLE_QUEUE) == FLUSH_SUBTITLE_QUEUE)
     {
-      if ((flags & FLUSH_SUBTITLE_QUEUE) == FLUSH_SUBTITLE_QUEUE)
+      subtitleLock.writeLock().lock();
+
+      try
       {
         subEntries.clear();
         subEntryPos = -1;
       }
-      return insertEntryForPostedInfo(time, dur, rawText);
+      finally
+      {
+        subtitleLock.writeLock().unlock();
+      }
     }
+
+    return insertEntryForPostedInfo(time, dur, rawText);
```

Comment:
Oh wait...that won't work because then you'll be asking for the lock again...could work around that with a parameter, but it's likely fine the way you have it.